plugins/ltac/tacinterp.ml

```diff
@@ -309,6 +309,23 @@ let extend_values_with_bindings (ln,lm) lfun =
   let accu = lfun +++ accu in
   Id.Map.fold (fun id c accu -> Id.Map.add id (of_cub c) accu) lm accu
 
+let flush_value_evars sigma v =
+  if has_type v (topwit wit_constr) then
+    let c = out_gen (topwit wit_constr) v in
+    let c =
+      try EConstr.of_constr (Evarutil.flush_and_check_evars sigma c)
+      with Evarutil.Uninstantiated_evar _ ->
+        (* How to tell to not use this binding anymore? *)
+        (* If it is used it might fail because of the evar *)
+        (* But this has to work if it is not used *)
```

Comment:
We could map to some special `wit`?

Reply:
You mean some `wit_unavailable` which triggers an error when we try to access it, or something like that?

Reply:
In any case, this wouldn't be a genarg but a toplevel tag value. And I don't think it would be a special value precisely because we want to share toplevel representations as much as possible.

Reply:
>You mean some wit_unavailable which triggers an error when we try to access it, or something like that?

yes
>In any case, this wouldn't be a genarg but a toplevel tag value. And I don't think it would be a special value precisely because we want to share toplevel representations as much as possible.

I don't know what any of this means.

Reply:
Whatever. Don't mess with the bound variable environment proactively, this would break some hidden invariants. This PR is a hackish workaround, instead we should make clients more robust. The problem is much more general than abstract, most functions should tolerate terms with missing evars or variables and fail gracefully instead of raising anomalies.